File: 3099.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        total = 0
        n = x

        while(n != 0):
            total += n % 10
            n //= 10

        if(x % total == 0):
            return total
        return -1

File: 3099.cpp
class Solution {
public:
    int sumOfTheDigitsOfHarshadNumber(int x) {
        int sum_of_digits = 0;
        int num = x;
        while(num != 0) {
            sum_of_digits += num % 10;
            num /= 10;
        }
        if(x % sum_of_digits == 0) {
            return sum_of_digits;
        }
        return -1;
    }
};